.github/workflows/inspektor-gadget.yml

```diff
@@ -240,6 +240,42 @@ jobs:
         cache-from: type=local,src=/tmp/.buildx-cache
         cache-to: type=local,dest=/tmp/.buildx-cache-new
 
+  build-examples:
+    name: Build examples
+    # level: 0
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        example: [runc-hook, kube-container-collection]
+    steps:
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Login to Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ${{ secrets.CONTAINER_REGISTRY }}
+        username: ${{ secrets.CONTAINER_REGISTRY_USERNAME }}
+        password: ${{ secrets.CONTAINER_REGISTRY_PASSWORD }}
+    - name: Check out code
+      uses: actions/checkout@v1
+    - name: Set IMAGE_TAG
+      run: |
+        TMP1=${GITHUB_REF#*/}
+        TMP2=${TMP1#*/}
+        IMAGE_TAG=${TMP2//\//-}
+        if [ "$IMAGE_TAG" = "main" ]; then
+            IMAGE_TAG="latest"
+        fi
+        echo IMAGE_TAG=$IMAGE_TAG >> $GITHUB_ENV
+    - name: Build example container
+      uses: docker/build-push-action@v2
+      with:
+        context: /home/runner/work/inspektor-gadget/inspektor-gadget
+        file: /home/runner/work/inspektor-gadget/inspektor-gadget/examples/${{ matrix.example }}/Dockerfile
+        push: true
```

Comment:
Do we really want to push all these example containers? 

Two concerns I have:
- The overhead of maintaining this. changes like https://github.com/kinvolk/inspektor-gadget/pull/597 will be more complicated because we'll have to consider more container images.
- Forks: I fear this would create further complication in forks if they don't set up all the container registries for these examples.